consensus/consensus_v2.go

```diff
@@ -277,6 +277,8 @@ func (consensus *Consensus) BlockCommitSigs(blockNum uint64) ([]byte, error) {
 		return nil, nil
 	}
 	lastCommits, err := consensus.Blockchain().ReadCommitSig(blockNum)
+	consensus.mutex.Lock()
+	defer consensus.mutex.Unlock()
```

Comment:
could you explain why we need this here ?

Reply:
this code is called from function `BlockCommitSigs` which is running in different goroutine. Without this lock it's going to be a race condition, because FBFTLog, which is called later, is not thread safe.